atlasdb-buggify/src/test/java/com/palantir/atlasdb/buggify/impl/DefaultBuggifyFactoryTest.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.buggify.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+
+public class DefaultBuggifyFactoryTest {
+
+    @Test
+    public void maybeReturnsDefaultWhenMaxProbability() {
+        assertThat(DefaultBuggifyFactory.INSTANCE.maybe(1.0)).isSameAs(DefaultBuggify.INSTANCE);
+    }
+
+    @Test
+    public void maybeReturnsNoOpWhenNeverProbability() {
+        assertThat(DefaultBuggifyFactory.INSTANCE.maybe(0.0)).isSameAs(NoOpBuggify.INSTANCE);
+    }
+
+    @Test
+    public void maybeReturnsNoOpWhenBelowOrEqualToProbability() {
+        assertThat(new DefaultBuggifyFactory(() -> 0.50).maybe(0.50)).isSameAs(NoOpBuggify.INSTANCE);
+    }
+
+    @Test
+    public void maybeReturnsDefaultWhenAboveProbability() {
+        assertThat(new DefaultBuggifyFactory(() -> 0.49).maybe(0.50)).isSameAs(DefaultBuggify.INSTANCE);
+    }
```

Comment:
That makes sense, although I'd still push for this approach, just because otherwise we're effectively testing `SecureRandom` ability to RNG. I agree it makes the test kind of wonky, as it's testing an implementation detail rather than an API to a degree. What do you think?

Reply:
I think we should keep these, to be clear, but also have the bigger test (yeah it would fail if `SecureRandom` is broken, but I would want our usage of it to be tested as well - and we could have some kind of spy to "test that we call secure random once" or something like that, but there's still a risk we flip the direction of the comparison or something like that, and I'm thinking setting up such expectations is probably harder than having the integration-y test).